libraries/rush-lib/src/cli/actions/CheckAction.ts

```diff
@@ -37,12 +38,21 @@ export class CheckAction extends BaseRushAction {
         'If this flag is specified, long lists of package names will not be truncated. ' +
         `This has no effect if the ${this._jsonFlag.longName} flag is also specified.`
     });
+    this._subspaceParameter = this.defineStringParameter({
+      parameterLongName: '--subspace',
+      argumentName: 'SUBSPACE_NAME',
+      description:
+        '(EXPERIMENTAL) Specifies a Rush subspace to be installed. Requires the "subspacesEnabled" feature to be enabled in subspaces.json.'
+    });
   }
 
   protected async runAsync(): Promise<void> {
     VersionMismatchFinder.rushCheck(this.rushConfiguration, this._terminal, {
       printAsJson: this._jsonFlag.value,
-      truncateLongPackageNameLists: !this._verboseFlag.value
+      truncateLongPackageNameLists: !this._verboseFlag.value,
+      subspace: this._subspaceParameter?.value
+        ? this.rushConfiguration.getSubspace(this._subspaceParameter.value)
+        : this.rushConfiguration.defaultSubspace
```

Comment:
Based on our chat, I think `rush check` must still enforce consistency across all projects.

The reason is that `ensureConsistentVersions` in **rush.json** must do that, because it has no way to say which subspace it applies to (we need a new setting for that).  Since `rush check` is conceptually associated with that setting, `rush check` needs some syntax that performs the same operation as `ensureConsistentVersions`.  Also, `rush install` installs all subspaces, not the `default` one.

Reply:
Changed to check consistency across all projects if no `--subspace` parameter is passed